tests/Spec.hs

```diff
@@ -145,142 +149,160 @@ defaultResults = Results
 -- Consume the head of the field with given getter and setter in the Results.
 takeFromList
   :: HasCallStack
-  => Monoid w
+  => State Results :> es
   => String
   -> (Results -> [a])
   -> ([a] -> Results -> Results)
-  -> RWS r w Results a
```

Comment:
`Spec.hs` used `mtl` for its `RWS` monad. It's a combination of Reader, Writer, State. Of these, the writer and state were used. Effectful has primitive effects for all three, so I just used those.